plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusClient.java

```diff
@@ -118,17 +122,34 @@ public PrometheusTable getTable(String schema, String tableName)
         if (tableNames == null) {
             return null;
         }
-        if (!tableNames.contains(tableName)) {
+
+        String remoteTableName = toRemoteTableName(tableName);
+        if (!tableNames.contains(remoteTableName)) {
             return null;
         }
+
         return new PrometheusTable(
-                tableName,
+                remoteTableName,
                 ImmutableList.of(
                         new PrometheusColumn("labels", varcharMapType),
                         new PrometheusColumn("timestamp", TIMESTAMP_COLUMN_TYPE),
                         new PrometheusColumn("value", DoubleType.DOUBLE)));
     }
 
+    private String toRemoteTableName(String tableName)
+    {
+        verify(tableName.equals(tableName.toLowerCase(ENGLISH)), "tableName not in lower-case: %s", tableName);
+        if (!caseInsensitiveNameMatching) {
+            return tableName;
+        }
+        for (String remoteTableName : (List<String>) tableSupplier.get().get("data")) {
```

Comment:
Tables names were already fetched at L121. Let's reuse it instead of reloading.

Reply:
OK, I will update the patch later, thanks~